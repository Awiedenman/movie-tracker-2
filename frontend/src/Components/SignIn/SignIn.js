import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class SignIn extends Component {

  render() {
    return (
      <section>
        <h1>Sign In</h1>
        <form action="">
          <input type="text" name="" placeholder="email" />
          <input type="password" name="" placeholder="password" />
          <button>Sign In</button>
        </form>
      </section>
    );
  }
}

export default SignIn;
