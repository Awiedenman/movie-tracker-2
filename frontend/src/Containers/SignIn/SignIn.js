import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = event => {
    const { name, value } = event.target;
    console.log(name);
    this.setState({ [name]: value });
  }

  handleSubmit = event => {
    event.preventDefault();
    // TODO make a POST request here

    this.setState({ email: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <section>
        <h1>Sign In</h1>
        <form action="POST" onSubmit={this.handleSubmit}>
          <input type="text" value={email} name="email" placeholder="email" onChange={this.handleChange} />
          <input type="password" value={password} name="password" placeholder="password" />
          <button>Sign In</button>
        </form>
      </section>
    );
  }
}

export default SignIn;
