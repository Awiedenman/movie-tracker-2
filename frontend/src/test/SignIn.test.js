import React from 'react';
import { shallow } from 'enzyme';
import SignIn from '../Components/SignIn/SignIn';

describe('SignIn', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<SignIn />));
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
