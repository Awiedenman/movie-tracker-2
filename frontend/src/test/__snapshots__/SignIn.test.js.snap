// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Sign In
</h1>,
        <form
          action=""
>
          <input
                    name=""
                    placeholder="email"
                    type="text"
          />
          <input
                    name=""
                    placeholder="password"
                    type="password"
          />
          <button>
                    Sign In
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <input
              name=""
              placeholder="email"
              type="text"
/>,
            <input
              name=""
              placeholder="password"
              type="password"
/>,
            <button>
              Sign In
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "",
              "placeholder": "email",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "",
              "placeholder": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Sign In
</h1>,
          <form
            action=""
>
            <input
                        name=""
                        placeholder="email"
                        type="text"
            />
            <input
                        name=""
                        placeholder="password"
                        type="password"
            />
            <button>
                        Sign In
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "action": "",
            "children": Array [
              <input
                name=""
                placeholder="email"
                type="text"
/>,
              <input
                name=""
                placeholder="password"
                type="password"
/>,
              <button>
                Sign In
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "",
                "placeholder": "email",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "",
                "placeholder": "password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
